galley/pkg/config/analysis/analyzers/serviceentry/protocoladresses.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package serviceentry
+
+import (
+	"fmt"
+
+	"istio.io/api/networking/v1alpha3"
+	"istio.io/istio/galley/pkg/config/analysis"
+	"istio.io/istio/galley/pkg/config/analysis/analyzers/util"
+	"istio.io/istio/galley/pkg/config/analysis/msg"
+	"istio.io/istio/pkg/config/resource"
+	"istio.io/istio/pkg/config/schema/collection"
+	"istio.io/istio/pkg/config/schema/collections"
+)
+
+type ProtocolAdressesAnalyzer struct{}
+
+var _ analysis.Analyzer = &ProtocolAdressesAnalyzer{}
+
+func (serviceEntry *ProtocolAdressesAnalyzer) Metadata() analysis.Metadata {
+	return analysis.Metadata{
+		Name:        "serviceentry.Analyzer",
+		Description: "Checks the validity of ServiceEntry",
+		Inputs: collection.Names{
+			collections.IstioNetworkingV1Alpha3Serviceentries.Name(),
+		},
+	}
+}
+
+func (serviceEntry *ProtocolAdressesAnalyzer) Analyze(context analysis.Context) {
+	context.ForEach(collections.IstioNetworkingV1Alpha3Serviceentries.Name(), func(resource *resource.Instance) bool {
+		serviceEntry.analyzeProtocolAddressesMissing(resource, context)
+		serviceEntry.analyzeProtocolTCPAddressesMissing(resource, context)
+		return true
+	})
+}
+
+func (serviceEntry *ProtocolAdressesAnalyzer) analyzeProtocolAddressesMissing(resource *resource.Instance, context analysis.Context) {
+	se := resource.Message.(*v1alpha3.ServiceEntry)
+
+	if se.Addresses == nil {
+		for index, port := range se.Ports {
+			if port.Protocol == "" {
+				message := msg.NewServiceEntryMissingAddressesAndProtocol(resource)
+
+				if line, ok := util.ErrorLine(resource, fmt.Sprintf(util.ServiceEntryPort, index)); ok {
+					message.Line = line
+				}
+
+				context.Report(collections.IstioNetworkingV1Alpha3Serviceentries.Name(), message)
+			}
+		}
+	}
+}
+
+func (serviceEntry *ProtocolAdressesAnalyzer) analyzeProtocolTCPAddressesMissing(resource *resource.Instance, context analysis.Context) {
+	se := resource.Message.(*v1alpha3.ServiceEntry)
+
+	if se.Addresses == nil {
+		for index, port := range se.Ports {
+			if port.Protocol == "TCP" {
```

Comment:
I think you can combine this condition with the above function. So there will be 2 checks.
one with `port.Protocol == ""` and other with `port.Protocol == "TCP"`?

Reply:
actually yes we can do that. do you have suggestions for the message in `galley/pkg/config/analysis/msg/messages.yaml`? cc @howardjohn

Reply:
@zufardhiyaulhaq To create a new message, edit _messages.yaml_.  Then run `go generate` in the messages directory after setting `$REPO_ROOT`.  I do this:

```
export REPO_ROOT=~/go/src/istio.io/istio
cd ~/go/src/istio.io/istio/galley/pkg/config/analysis/msg
go generate
```

Reply:
@esnible pardon me, not creating a new message. I mean the message format. currently, I am using two analyzer code (IST0134 and IST0135) to separate
- empty `Addresses` and empty `Protocol`
- empty `Addresses` with `Protocol` TCP

@shamsher31 this is done